lib/datasource/gitlab-packages/index.ts

```diff
@@ -0,0 +1,81 @@
+import { cache } from '../../util/cache/package/decorator';
+import { GitlabHttp } from '../../util/http/gitlab';
+import * as url from '../../util/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { datasource } from './common';
+import type { GitlabPackage } from './types';
+
+// Gitlab Packages API: https://docs.gitlab.com/ee/api/packages.html
+
+export class GitlabPackagesDatasource extends Datasource {
+  static readonly id = datasource;
+
+  protected override http: GitlabHttp;
+
+  override caching = true;
+
+  override customRegistrySupport = true;
+
+  override defaultVersioning = 'docker';
+
+  constructor() {
+    super(datasource);
+    this.http = new GitlabHttp();
+  }
+
+  static getGitlabPackageApiUrl(
+    registryUrl: string,
+    lookupName: string
+  ): string {
+    const parsedRegistryUrl = url.parseUrl(registryUrl);
+
+    const packageName = encodeURIComponent(lookupName);
+
+    const server = parsedRegistryUrl.origin;
+    const project = encodeURIComponent(parsedRegistryUrl.pathname.substring(1)); // remove leading /
+
+    return url.resolveBaseUrl(
+      server,
+      `/api/v4/projects/${project}/packages?package_name=${packageName}&per_page=100`
+    );
```

Comment:
I would prefer that the registry url is the path to gitlab only and use `lookupName` as project.

So i can host gitlab in a sub path, eg `https://domain.test/gitlab` won't currently work

Reply:
also use use `url.joinUrlParts` instead of `resolveBaseUrl`